server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -308,28 +342,142 @@ private void takeSnapshot(Map<String, LookupExtractorFactoryContainer> lookupMap
+                    LOG.info(
+                        "Starting lookup [%s]:[%s]",
+                        lookupBean.getName(),
+                        container
+                    );
+                    if (container.getLookupExtractorFactory().start()) {
+                      LOG.info(
+                          "Started lookup [%s]:[%s]",
+                          lookupBean.getName(),
+                          container
+                      );
+                      return new AbstractMap.SimpleImmutableEntry<>(lookupBean.getName(), container);
+                    } else {
+                      LOG.error(
+                          "Failed to start lookup [%s]:[%s]",
+                          lookupBean.getName(),
+                          container
+                      );
+                      return null;
+                    }
+                  }
+              )
+          );
+        }
+        final ListenableFuture<List<Map.Entry>> futureList = Futures.allAsList(futures);
+        try {
+          futureList.get()
+                    .stream()
+                    .filter(Objects::nonNull)
+                    .forEach(builder::put);
+
+          stateRef.set(new LookupUpdateState(builder.build(), ImmutableList.of(), ImmutableList.of()));
         }
+        catch (Exception ex) {
+          futureList.cancel(true);
+          throw ex;
+        }
+      }
+      catch (Exception e) {
+        LOG.error(e, "Failed to finish lookup load process.");
       }
+      finally {
+        try {
+          executorService.shutdownNow();
+          if (!executorService.awaitTermination(60, TimeUnit.SECONDS)) {
+            LOG.warn("Lookup loading didn't complete in 60 seconds!");
+          }
+        }
+        catch (InterruptedException e) {
+          LOG.error("Lookup loading interrupted. ");
+          Thread.currentThread().interrupt();
 
-      stateRef.set(new LookupUpdateState(builder.build(), ImmutableList.of(), ImmutableList.of()));
+        }
+      }
     } else {
+      LOG.info("No lookups to be loaded at this point");
       stateRef.set(new LookupUpdateState(ImmutableMap.of(), ImmutableList.of(), ImmutableList.of()));
     }
   }
 
+  private List<LookupBean> getLookupListFromCoordinator(String tier)
+  {
+    try {
+      final FullResponseHolder response = fetchLookupsForTier(tier);
+      List<LookupBean> lookupBeanList = new ArrayList<>();
+      if (!response.getStatus().equals(HttpResponseStatus.OK)) {
+        LOG.error(
+            "Error while fetching lookup code from Coordinator with status[%s] and content[%s]",
+            response.getStatus(),
+            response.getContent()
+        );
+        return null;
+      }
+
+      // Older version of getSpecificTier returns a list of lookup names.
+      // Lookup loading is performed via snapshot if older version is present.
+      // This check is only for backward compatibility and should be removed in a future release
```

Comment:
Could you please create a Github issue about this? With a recommended milestone version

Reply:
#4875